akka-stream/src/main/scala/akka/stream/impl/streamref/SinkRefImpl.scala

```diff
@@ -136,7 +148,8 @@ private[stream] final class SinkRefStageImpl[In] private[akka] (
             s"[$stageActorName] Remote side did not subscribe (materialize) handed out Source reference [${promise.future.value}], " +
               s"within subscription timeout: ${PrettyDuration.format(subscriptionTimeout.timeout)}!")
 
-          throw ex // this will also log the exception, unlike failStage; this should fail rarely, but would be good to have it "loud"
+          getPartnerRef ! StreamRefsProtocol.RemoteStreamFailure(ex.getMessage)
+          finishedWithAwaitingPartnerTermination = OptionVal(Failure(ex))
```

Comment:
what is triggering the final tear down in this SubscriptionTimeout case?
We act on the `finishedWithAwaitingPartnerTermination` when receiving `Terminated`, but we only watch after receiving the partner. How is that related to the subscription timeout?

Reply:
this was a real bug, thanks for finding -- it has to throw